src/senselab/audio/tasks/forced_alignment/forced_alignment.py

```diff
@@ -0,0 +1,747 @@
+"""Align function based on WhisperX implementation."""
+
+import math
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+import torch
+from nltk.tokenize.punkt import PunktParameters, PunktSentenceTokenizer
+from threadpoolctl import threadpool_limits
+from transformers import Wav2Vec2ForCTC, Wav2Vec2Processor
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.audio.tasks.forced_alignment.constants import (
+    DEFAULT_ALIGN_MODELS_HF,
+    LANGUAGES_WITHOUT_SPACES,
+    PUNKT_ABBREVIATIONS,
+    SAMPLE_RATE,
+)
+from senselab.audio.tasks.forced_alignment.data_structures import (
+    AlignedTranscriptionResult,
+    Point,
+    Segment,
+    SingleAlignedSegment,
+    SingleSegment,
+    SingleWordSegment,
+)
+from senselab.audio.tasks.preprocessing.preprocessing import resample_audios
+from senselab.utils.data_structures.device import DeviceType, _select_device_and_dtype
+from senselab.utils.data_structures.language import Language
+from senselab.utils.data_structures.script_line import ScriptLine
+
+
+def _prepare_audio(audio: Audio) -> Audio:
```

Comment:
why this method? our Audio objects always contain waveform which is a tensor of shape (channels, timestamps)

Reply:
Removed.